sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileShareTestBase.java

```diff
@@ -0,0 +1,674 @@
+import com.azure.storage.file.share.specialized.ShareLeaseClientBuilder;
+import okhttp3.ConnectionPool;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.zip.CRC32;
+
+
+public class FileShareTestBase extends TestProxyTestBase {
+    protected static final TestEnvironment ENVIRONMENT = TestEnvironment.getInstance();
+    protected static final TestDataFactory DATA = TestDataFactory.getInstance();
+    private static final HttpClient NETTY_HTTP_CLIENT = new NettyAsyncHttpClientBuilder().build();
+    private static final HttpClient OK_HTTP_CLIENT = new OkHttpAsyncHttpClientBuilder()
+        .connectionPool(new ConnectionPool(50, 5, TimeUnit.MINUTES))
+        .build();
+
+    protected static final HttpHeaderName X_MS_VERSION = HttpHeaderName.fromString("x-ms-version");
+    protected static final HttpHeaderName X_MS_REQUEST_ID = HttpHeaderName.fromString("x-ms-request-id");
+
+    private static final ClientLogger LOGGER = new ClientLogger(FileShareTestBase.class);
+
+    protected String prefix;
+
+    private int entityNo = 0; // Used to generate stable share names for recording tests requiring multiple shares.
+
+    /*
+    Note that this value is only used to check if we are depending on the received etag. This value will not actually
+    be used.
+    */
+    protected static final String RECEIVED_LEASE_ID = "received";
+    static final String GARBAGE_LEASE_ID = UUID.randomUUID().toString();
+
+    URL testFolder = getClass().getClassLoader().getResource("testfiles");
+
+
+    // Clients for API tests
+    protected ShareServiceClient primaryFileServiceClient;
+    protected ShareServiceAsyncClient primaryFileServiceAsyncClient;
+    protected ShareServiceClient premiumFileServiceClient;
+    protected ShareServiceAsyncClient premiumFileServiceAsyncClient;
+
+    @Override
+    public void beforeTest() {
+        super.beforeTest();
+        prefix = getCrc32(testContextManager.getTestPlaybackRecordingName());
+
+        if (getTestMode() != TestMode.LIVE) {
+            interceptorManager.addSanitizers(Arrays.asList(
+                new TestProxySanitizer("sig=(.*)", "REDACTED", TestProxySanitizerType.URL),
+                new TestProxySanitizer("x-ms-file-rename-source", "sig=(.*)", "REDACTED", TestProxySanitizerType.HEADER),
+                new TestProxySanitizer("x-ms-copy-source", "sig=(.*)", "REDACTED", TestProxySanitizerType.HEADER)));
```

Comment:
Need to add `x-ms-copy-source-authorization` here to redact the bearer token. I don't know if it exists but if it does also add `x-ms-file-rename-source-authorization`.